lib/continents_finder.rb

```diff
@@ -59,11 +59,13 @@ def positions_around_of(land)
   def add_to_continent(land, continent)
     land.continent = continent
     positions_around_of(land).each do |position|
-      @lands.each do |l|
-        if !l.belongs_to_a_continent? && position[:x] == l.x && position[:y] == l.y
-          add_to_continent(l, continent)
-        end
-      end
+      land_to_add = unmarked_land_in(position)
+      add_to_continent(land_to_add, continent) if land_to_add
     end
   end
+
+  def unmarked_land_in(position)
+    land = @lands.find { |l| position[:x] == l.x && position[:y] == l.y }
+    land && !land.belongs_to_a_continent? ? land : nil
+  end
```

Comment:
This method is probably need a comment - wasn't as clear now as I read it 8 days later. 
The method tries to ascertain if there is an unmarked land on a coordinate.